internal/transform/config_setting.go

```diff
@@ -0,0 +1,164 @@
+package transform
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/stellar/go/ingest"
+	"github.com/stellar/go/xdr"
+	"github.com/stellar/stellar-etl/internal/utils"
+)
+
+// TransformConfigSetting converts an liquidity pool ledger change entry into a form suitable for BigQuery
+func TransformConfigSetting(ledgerChange ingest.Change) (ConfigSettingOutput, error) {
+	ledgerEntry, changeType, outputDeleted, err := utils.ExtractEntryFromChange(ledgerChange)
+	if err != nil {
+		return ConfigSettingOutput{}, err
+	}
+
+	// LedgerEntryChange must contain a liquidity pool state change to be parsed, otherwise skip
+	if ledgerEntry.Data.Type != xdr.LedgerEntryTypeConfigSetting {
+		return ConfigSettingOutput{}, nil
+	}
+
+	configSetting, ok := ledgerEntry.Data.GetConfigSetting()
+	if !ok {
+		return ConfigSettingOutput{}, fmt.Errorf("Could not extract config setting from ledger entry; actual type is %s", ledgerEntry.Data.Type)
+	}
+
+	configSettingId := configSetting.ConfigSettingId
+
+	contractMaxSizeBytes, ok := configSetting.GetContractMaxSizeBytes()
+
+	contractCompute, ok := configSetting.GetContractCompute()
```

Comment:
do we need to catch any errors if they occur? Would it be better to use a `MustContractMaxSizeBytes`?

Reply:
I believe the problem with the Must* functions is they `panic` which basically stops everything.
I removed the checks because most of the time most fields won't be set. Maybe I could refactor this to more like a switch statement with a struct of default null/zero values.